src/command_modules/azure-cli-keyvault/azure/cli/command_modules/keyvault/_validators.py

```diff
@@ -197,8 +197,9 @@ def base64_encoded_certificate_type(string):
     return cert_data
 
 def datetime_type(string):
-    """ Validates UTC datettime in format '%Y-%m-%d\'T\'%H:%M\'Z\''. """
-    date_format = '%Y-%m-%dT%H:%MZ'
+    ''' Validates UTC datettime in format '%Y-%m-%d\'T\'%H:%M:%S\'Z\''
+        or  '%Y-%m-%d\'T\'%H:%M\'Z\''. '''
+    date_format = '%Y-%m-%dT%H:%M:%SZ' if len(string) is 20 else '%Y-%m-%dT%H:%MZ'
```

Comment:
What is basis for `if len(string) is 20`?  That seems like a magic number.
It might be better to simply try strptime with the more extensive date format, catch the parse error and then try the less limiting one.

Also, we should make sure we can still parse times with missing components. For example default minutes or seconds to 0 if they aren't provided (like the old strings). This we should be able to verify with a set of robust unit tests.